### PR TITLE
[docs] document new custom builds features

### PR DESCRIPTION
**Pull request by dsokal**
# Why

We recently released a few new features for custom builds:
- Importing functions from other files.
- Limiting functions to run only on supported platforms. (If someone attempts to run it on an unsupported one, it'll throw an error).
- Checking out the project directory. This is a breaking change! If you don't call `eas/checkout`, you don't have access to the project sources and your workflow works with an empty working directory.

# How

Document the new features.

# Test Plan

`yarn dev`

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 1]
```diff
@@ -463,18 +466,92 @@ functions:
...
+
+```yaml
+functions:
+  greetings:
+    name: Say Hi!
+    # @info #
+    supported_platforms: [darwin]
+    # @end #
+    command: echo "Hi!"
+```
```
Comment (szdziedzic) [inline docs/pages/preview/custom-build-config-schema.mdx:486]: Thanks!

**Review by szdziedzic — APPROVED** [review]
Nice! 🚀

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 2]
```diff
@@ -463,18 +466,92 @@ functions:
...
-EAS provides a built-in reusable function called `upload_artifact` that you can use in a workflow without defining the function definition.
+EAS provides a set of built-in reusable functions that you can use in a workflow without defining the function definition.
+
+> **info** **Tip:** Any function that is built-in and provided by EAS must start with the prefix `eas/`.
+
+#### `eas/checkout`
+
+Checks out you project source files.
+
+For example, a workflow with the following `steps` will check out the project and list the files in the `assets` directory:
```
Comment (amandeepmittal) [inline docs/pages/preview/custom-build-config-schema.mdx:531]:
```suggestion
For example, a workflow with the following `steps` will check out the project and list the files in the **assets** directory:
```

file/directory names to be bold for consistency as per our writing styelguide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text

**Review by amandeepmittal — APPROVED** [review]
Thanks for making these updates! Left one small suggestion.

**Pull request merged**